packages/core/src/components/chat.ts

```diff
@@ -41,83 +41,87 @@ export type ChatOptions = {
-const encoder = new TextEncoder();
-const decoder = new TextDecoder();
+const topicSubjectMap: WeakMap<Room, Map<string, Subject<ReceivedChatMessage>>> = new WeakMap();
 
-const topicSubjectMap: Map<Room, Map<string, Subject<RawMessage>>> = new Map();
+function isIgnorableChatMessage(msg: ReceivedChatMessage | LegacyReceivedChatMessage) {
+  return (msg as LegacyChatMessage).ignoreLegacy == true;
+}
 
-const encode = (message: LegacyReceivedChatMessage) => encoder.encode(JSON.stringify(message));
+const decodeLegacyMsg = (message: Uint8Array) =>
+  JSON.parse(new TextDecoder().decode(message)) as LegacyReceivedChatMessage | ReceivedChatMessage;
 
-const decode = (message: Uint8Array) =>
-  JSON.parse(decoder.decode(message)) as LegacyReceivedChatMessage | ReceivedChatMessage;
+const encodeLegacyMsg = (message: LegacyReceivedChatMessage) =>
+  new TextEncoder().encode(JSON.stringify(message));
 
 export function setupChat(room: Room, options?: ChatOptions) {
-  const onDestroyObservable = new Subject<void>();
-
-  const serverSupportsChatApi = () =>
+  const serverSupportsDataStreams = () =>
     room.serverInfo?.edition === 1 ||
-    (!!room.serverInfo?.version && compareVersions(room.serverInfo?.version, '1.17.2') > 0);
-
-  const { messageDecoder, messageEncoder, channelTopic, updateChannelTopic } = options ?? {};
+    (!!room.serverInfo?.version && compareVersions(room.serverInfo?.version, '1.8.2') > 0);
 
-  const topic = channelTopic ?? DataTopic.CHAT;
+  const onDestroyObservable = new Subject<void>();
 
-  const updateTopic = updateChannelTopic ?? DataTopic.CHAT_UPDATE;
+  const topic = options?.channelTopic ?? DataTopic.CHAT;
+  const legacyTopic = options?.channelTopic ?? LegacyDataTopic.CHAT;
 
   let needsSetup = false;
   if (!topicSubjectMap.has(room)) {
     needsSetup = true;
   }
-  const topicMap = topicSubjectMap.get(room) ?? new Map<string, Subject<RawMessage>>();
-  const messageSubject = topicMap.get(topic) ?? new Subject<RawMessage>();
+  const topicMap = topicSubjectMap.get(room) ?? new Map<string, Subject<ReceivedChatMessage>>();
+  const messageSubject = topicMap.get(topic) ?? new Subject<ReceivedChatMessage>();
   topicMap.set(topic, messageSubject);
   topicSubjectMap.set(room, topicMap);
 
+  const finalMessageDecoder = options?.messageDecoder ?? decodeLegacyMsg;
   if (needsSetup) {
-    /** Subscribe to all appropriate messages sent over the wire. */
-    const { messageObservable } = setupDataMessageHandler(room, [topic, updateTopic]);
-    messageObservable.pipe(takeUntil(onDestroyObservable)).subscribe(messageSubject);
+    room.registerTextStreamHandler(topic, async (reader, participantInfo) => {
+      const { id, timestamp } = reader.info;
+      const streamObservable = from(reader).pipe(
+        scan((acc: string, chunk: string) => {
+          return acc + chunk;
+        }),
+        map((chunk: string) => {
+          console.log('text stream updated', { id, chunk, attributes: reader.info.attributes });
```

Comment:
should these console.logs be removed or downgraded to debug?

Reply:
thanks for catching that, removed